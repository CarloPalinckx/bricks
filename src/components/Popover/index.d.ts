import { Component } from 'react';
import { PopperChildrenProps } from 'react-popper';
declare type PlacementType = PopperChildrenProps['placement'];
declare type PropsType = {
    placement?: PlacementType;
    isOpen?: boolean;
    fixed?: boolean;
    offset?: number;
    distance?: number;
    stretch?: boolean;
    triggerOn?: 'click' | 'hover';
    renderContent(): JSX.Element | string;
};
declare type StateType = {
    isOpen: boolean;
};
declare class Popover extends Component<PropsType, StateType> {
    private anchorRef;
    private popoverRef;
    constructor(props: PropsType);
    static getDerivedStateFromProps(props: PropsType, state: StateType): Partial<StateType>;
    private mapOffset;
    private togglePopover;
    private handleClickOutside;
    componentDidMount(): void;
    componentWillUnmount(): void;
    render(): JSX.Element;
}
export default Popover;
export { PropsType, PlacementType };
