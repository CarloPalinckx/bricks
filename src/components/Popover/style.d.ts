import _R from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import { PlacementType } from '.';
import _T from '../../types/ThemeType';
declare type PopoverThemeType = {
    background: string;
    borderRadius: string;
};
declare type PopoverAnchorPropsType = {
    stretch?: boolean;
};
declare const PopoverAnchor: _S<PopoverAnchorPropsType & _R.HTMLProps<HTMLDivElement>, _T, PopoverAnchorPropsType & _R.HTMLProps<HTMLDivElement>>;
declare const PopoverBackground: _S<_R.DetailedHTMLProps<_R.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, _T, _R.DetailedHTMLProps<_R.HTMLAttributes<HTMLDivElement>, HTMLDivElement>>;
declare const PopoverContent: _S<_R.DetailedHTMLProps<_R.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, _T, _R.DetailedHTMLProps<_R.HTMLAttributes<HTMLDivElement>, HTMLDivElement>>;
declare const PopoverWindow: _S<_R.DetailedHTMLProps<_R.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, _T, _R.DetailedHTMLProps<_R.HTMLAttributes<HTMLDivElement>, HTMLDivElement>>;
declare type PopoverArrowPropsType = {
    placement: PlacementType;
    shadow?: boolean;
};
declare const PopoverArrow: _S<PopoverArrowPropsType & _R.HTMLProps<HTMLDivElement>, _T, PopoverArrowPropsType & _R.HTMLProps<HTMLDivElement>>;
export { PopoverAnchor, PopoverArrow, PopoverBackground, PopoverContent, PopoverWindow, PopoverThemeType };
