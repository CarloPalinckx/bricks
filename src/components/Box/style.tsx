import _R from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import { PropsType } from '.';
import _T from '../../types/ThemeType';
import styled, { withProps } from '../../utility/styled';

type BoxPropsType = PropsType & {
    flexWrap?: PropsType['wrap'];
    elementHeight?: PropsType['height'];
    flexDirection?: PropsType['direction'];
    flexOrder?: PropsType['order'];
};

const StyledBox = withProps<BoxPropsType, HTMLDivElement>(styled.div) `
    display: flex;
    height: ${({ elementHeight }): string =>
        elementHeight !== undefined ? `${elementHeight}` : 'auto'};
    flex-wrap: ${({ flexWrap }): string =>
        flexWrap !== undefined && flexWrap ? 'wrap' : 'no-wrap'};
    flex-direction: ${({ flexDirection }): string =>
        flexDirection !== undefined ? flexDirection : 'row'};
    justify-content: ${({ justifyContent }): string =>
        justifyContent !== undefined ? justifyContent : 'flex-start'};
    align-items: ${({ alignItems }): string =>
        alignItems !== undefined ? alignItems : 'stretch'};
    align-content: ${({ alignContent }): string =>
        alignContent !== undefined ? alignContent : 'stretch'};
    margin: ${({ margin }): string =>
        margin !== undefined
            ? `${margin.top} ${margin.right} ${margin.bottom} ${margin.left}`
            : '0'};
    flex-grow: ${({ grow }): number => (grow ? grow : 0)};
    flex-shrink: ${({ shrink }): number => (shrink ? shrink : 1)};
    flex-basis: ${({ basis }): string => (basis ? basis : 'auto')};
    order: ${({ flexOrder }): number => (flexOrder ? flexOrder : 0)};
    align-self: ${({ alignSelf }): string => (alignSelf ? alignSelf : 'auto')};
`;

export default StyledBox;
export { BoxPropsType };
