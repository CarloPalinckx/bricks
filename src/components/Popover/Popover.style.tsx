import {  ClassAttributes as _C, HTMLAttributes as _H } from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../types/ThemeType';
import styled from '../../utility/styled';
import { PopoverArrowPropsType } from './Popover.template';

const PopoverAnchor = styled.div`
    display: inline-block;
`;

const PopoverBackground = styled.div`
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    z-index: 2;
    box-shadow: 0 1px 12px rgba(0,0,0,.2);
    background: ${({ theme }):string => theme.Popover.background};
    border-radius: ${({ theme }):number => theme.Popover.borderRadius}px;
`;

const PopoverContent = styled.div`
    position: relative;
    z-index: 4;
`;

const PopoverWindow = styled.div`
    max-width: 500px;
`;

const PopoverArrow = styled.div`
    position: absolute;
    width: 16px;
    height: 16px;
    background: ${({ theme }):string => theme.Popover.background};

    ${({ placement }:PopoverArrowPropsType):string => {
        const pointerSize:number = 16;

        switch (placement) {
            case 'left':
            case 'left-start':
            case 'left-end':
                return `
                    right: ${-pointerSize / 2}px;
                    transform: scaleX(.75) rotate(45deg);
                `;
            case 'right':
            case 'right-start':
            case 'right-end':
                return `
                    left: ${-pointerSize / 2}px;
                    transform: scaleX(.75) rotate(45deg);
                `;
            case 'top':
            case 'top-start':
            case 'top-end':
                return `
                    bottom: ${-pointerSize / 2}px;
                    transform: scaleY(.75) rotate(45deg);
                `;
            default:
                return `
                    top: ${-pointerSize / 2}px;
                    transform: scaleY(.75) rotate(45deg);
                `;
        }
    }}

    ${({ theme, shadow }):string => shadow
        ? `
            box-shadow: 0 1px 12px rgba(0 , 0, 0, .2);
            z-index: 1;
        ` : `
            background: ${theme.Popover.background};
            z-index: 3;
        `
    }
`;

export {
    PopoverAnchor,
    PopoverArrow,
    PopoverBackground,
    PopoverContent,
    PopoverWindow,
};
