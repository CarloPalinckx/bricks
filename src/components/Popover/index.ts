export { default as default, PopoverThemeType } from './Popover.template';
