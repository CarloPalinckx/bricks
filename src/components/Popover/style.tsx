import _R from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import { PlacementType } from '.';
import _T from '../../types/ThemeType';
import styled, { withProps } from '../../utility/_styled';

type PopoverThemeType = {
    background: string;
    borderRadius: string;
};

type PopoverAnchorPropsType = {
    stretch?: boolean;
};

const PopoverAnchor = withProps<PopoverAnchorPropsType, HTMLDivElement>(styled.div)`
    ${({ stretch }): string =>
        stretch
            ? `
            display: block;
            flex-grow: 1;
            height: 100%;
        `
            : `
            display: inline-block;
        `}
`;

const PopoverBackground = styled.div`
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    z-index: 2;
    box-shadow: 0 3px 48px rgba(0, 0, 0, 0.3);
    background: ${({ theme }): string => theme.Popover.background};
    border-radius: ${({ theme }): string => theme.Popover.borderRadius};
`;

const PopoverContent = styled.div`
    position: relative;
    z-index: 4;
`;

const PopoverWindow = styled.div`
    max-width: 500px;
`;

type PopoverArrowPropsType = {
    placement: PlacementType;
    shadow?: boolean;
};

const PopoverArrow = withProps<PopoverArrowPropsType, HTMLDivElement>(styled.div)`
    position: absolute;
    width: 16px;
    height: 16px;
    background: ${({ theme }): string => theme.Popover.background};

    ${({ placement }): string => {
        const pointerSize: number = 16;

        switch (placement) {
            case 'left':
            case 'left-start':
            case 'left-end':
                return `
                    right: ${-pointerSize / 2}px;
                    transform: scaleX(.75) rotate(45deg);
                `;
            case 'right':
            case 'right-start':
            case 'right-end':
                return `
                    left: ${-pointerSize / 2}px;
                    transform: scaleX(.75) rotate(45deg);
                `;
            case 'top':
            case 'top-start':
            case 'top-end':
                return `
                    bottom: ${-pointerSize / 2}px;
                    transform: scaleY(.75) rotate(45deg);
                `;
            default:
                return `
                    top: ${-pointerSize / 2}px;
                    transform: scaleY(.75) rotate(45deg);
                `;
        }
    }} ${({ theme, shadow }): string =>
    shadow
        ? `
            box-shadow: 0 1px 12px rgba(0 , 0, 0, .2);
            z-index: 1;
        `
        : `
            background: ${theme.Popover.background};
            z-index: 3;
        `};
`;

export { PopoverAnchor, PopoverArrow, PopoverBackground, PopoverContent, PopoverWindow, PopoverThemeType };
