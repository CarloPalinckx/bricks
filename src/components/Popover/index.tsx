import React, { StatelessComponent } from 'react';
import { Manager, Popper, PopperChildrenProps, Reference, ReferenceChildrenProps } from 'react-popper';
import TransitionAnimation from '../TransitionAnimation';
import { PopoverAnchor, PopoverArrow, PopoverBackground, PopoverContent, PopoverWindow } from './style';

type PlacementType = PopperChildrenProps['placement'];

type PropsType = {
    placement?: PlacementType;
    isOpen: boolean;
    fixed?: boolean;
    offset?: number;
    renderContent(): JSX.Element | string;
};

const Popover: StatelessComponent<PropsType> = (props): JSX.Element => {
    return (
        <>
            <Manager>
                <Reference>
                    {({ ref }: ReferenceChildrenProps): JSX.Element => (
                        <PopoverAnchor innerRef={ref}>{props.children}</PopoverAnchor>
                    )}
                </Reference>
                <TransitionAnimation show={props.isOpen} animation="fade">
                    <Popper
                        positionFixed={props.fixed !== undefined ? props.fixed : false}
                        placement={props.placement !== undefined ? props.placement : 'bottom'}
                        modifiers={{
                            offset: {
                                offset: props.offset !== undefined ? `0 ${props.offset}px` : '0 16px',
                            },
                        }}
                    >
                        {({ ref, style, placement, arrowProps }: PopperChildrenProps): JSX.Element => (
                            <PopoverWindow innerRef={ref} style={style}>
                                <PopoverContent>{props.renderContent()}</PopoverContent>
                                <PopoverBackground />
                                <PopoverArrow
                                    innerRef={arrowProps.ref}
                                    style={arrowProps.style}
                                    placement={placement}
                                />
                                <PopoverArrow shadow style={arrowProps.style} placement={placement} />
                            </PopoverWindow>
                        )}
                    </Popper>
                </TransitionAnimation>
            </Manager>
        </>
    );
};

export default Popover;
export { PropsType, PlacementType };
