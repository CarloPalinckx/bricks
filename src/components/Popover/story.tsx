import { boolean, number, select } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import React, { Component, StatelessComponent } from 'react';
import Popover, { PlacementType } from '.';
import trbl from '../../utility/trbl';
import Box from '../Box';
import Button from '../Button';
import Text from '../Text';

type PropsType = {
    placement: PlacementType;
    fixed: boolean;
    offset: number;
    distance: number;
};

type StateType = {
    isOpen: boolean;
};

const DemoContent: StatelessComponent = (): JSX.Element => {
    return (
        <Box margin={trbl(24)}>
            <Text>
                Donec sed odio dui. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Maecenas faucibus
                mollis interdum. Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus
                commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.
            </Text>
        </Box>
    );
};

class Demo extends Component<PropsType, StateType> {
    public constructor(props: PropsType) {
        super(props);
        this.state = {
            isOpen: false,
        };
    }

    private toggle = (): void => {
        this.setState({
            isOpen: !this.state.isOpen,
        });
    };

    public render(): JSX.Element {
        return (
            <Box height="90vh" justifyContent="center" alignItems="center">
                <Box margin={trbl(48)}>
                    <Popover
                        isOpen={this.state.isOpen}
                        placement={this.props.placement}
                        fixed={this.props.fixed}
                        offset={this.props.offset}
                        distance={this.props.distance}
                        renderContent={(): JSX.Element => <DemoContent />}
                    >
                        <Button variant="primary" title="Toggle" action={this.toggle} />
                    </Popover>
                </Box>
            </Box>
        );
    }
}

storiesOf('Popover', module)
    .add('Default', () => (
        /* tslint:disable */
        <Demo
            placement={
                select(
                    'placement',
                    [
                        'auto-start',
                        'auto',
                        'auto-end',
                        'top-start',
                        'top',
                        'top-end',
                        'right-start',
                        'right',
                        'right-end',
                        'bottom-end',
                        'bottom',
                        'bottom-start',
                        'left-end',
                        'left',
                        'left-start',
                    ],
                    'bottom',
                ) as PlacementType
            }
            fixed={boolean('fixed', false)}
            offset={number('offset', 0)}
            distance={number('distance', 16)}
        />
        /* tslint:enable */
    ))
    .add('Internal state on hover', () => (
        <Box height="90vh" justifyContent="center" alignItems="center">
            <Box margin={trbl(48)}>
                <Popover
                    triggerOn={'hover'}
                    renderContent={(): JSX.Element => <DemoContent />}
                    placement={
                        select(
                            'placement',
                            [
                                'auto-start',
                                'auto',
                                'auto-end',
                                'top-start',
                                'top',
                                'top-end',
                                'right-start',
                                'right',
                                'right-end',
                                'bottom-end',
                                'bottom',
                                'bottom-start',
                                'left-end',
                                'left',
                                'left-start',
                            ],
                            'bottom',
                        ) as PlacementType
                    }
                    fixed={boolean('fixed', false)}
                    offset={number('offset', 0)}
                    distance={number('distance', 16)}
                >
                    <Button variant="primary" title="Hover over me" />
                </Popover>
            </Box>
        </Box>
    ))
    .add('Internal state on click', () => (
        <Box height="90vh" justifyContent="center" alignItems="center">
            <Box margin={trbl(48)}>
                <Popover
                    triggerOn={'click'}
                    renderContent={(): JSX.Element => <DemoContent />}
                    placement={
                        select(
                            'placement',
                            [
                                'auto-start',
                                'auto',
                                'auto-end',
                                'top-start',
                                'top',
                                'top-end',
                                'right-start',
                                'right',
                                'right-end',
                                'bottom-end',
                                'bottom',
                                'bottom-start',
                                'left-end',
                                'left',
                                'left-start',
                            ],
                            'bottom',
                        ) as PlacementType
                    }
                    fixed={boolean('fixed', false)}
                    offset={number('offset', 0)}
                    distance={number('distance', 16)}
                >
                    <Button variant="primary" title="Toggle" />
                </Popover>
            </Box>
        </Box>
    ));
