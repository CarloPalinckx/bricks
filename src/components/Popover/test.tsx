import toJson from 'enzyme-to-json';
import React from 'react';
import { Popper, Reference } from 'react-popper';
import Popover from '.';
import { shallowWithTheme, mountWithTheme } from '../../utility/_styled/testing';
import TransitionAnimation from '../TransitionAnimation';
import { PopoverAnchor, PopoverArrow, PopoverBackground } from './style';

describe('Popover', () => {
    it('should render with defaults', () => {
        const component = shallowWithTheme(<Popover isOpen={true} renderContent={(): string => 'Mock content'} />);

        const reference = component.find(Reference);
        const popper = component.find(Popper);

        expect(popper.prop('placement')).toEqual('bottom');
        expect(popper.prop('positionFixed')).toEqual(false);
        expect(toJson(popper.dive())).toMatchSnapshot();
        expect(toJson(reference.dive())).toMatchSnapshot();
    });

    it('should render closed', () => {
        const component = shallowWithTheme(<Popover isOpen={false} renderContent={(): string => 'Mock content'} />);

        const transition = component.find(TransitionAnimation);

        expect(transition.prop('show')).toEqual(false);
    });

    it('should render with a fixed postition', () => {
        const component = shallowWithTheme(
            <Popover isOpen={true} placement="left" fixed={true} renderContent={(): string => 'Mock content'} />,
        );

        const popper = component.find(Popper);

        expect(popper.prop('positionFixed')).toEqual(true);
    });

    it('should render with a custom distance and offset', () => {
        const component = shallowWithTheme(
            <Popover isOpen={true} offset={20} distance={6} renderContent={(): string => 'Mock content'} />,
        );

        const popper = component.find(Popper);

        expect(popper.prop('modifiers')).toEqual({
            offset: {
                offset: '20px, 6px',
            },
            flip: {
                enabled: false,
            },
        });
    });

    it('should render with only a custom offset', () => {
        const component = shallowWithTheme(
            <Popover isOpen={true} offset={20} renderContent={(): string => 'Mock content'} />,
        );

        const popper = component.find(Popper);

        expect(popper.prop('modifiers')).toEqual({
            offset: {
                offset: '20px, 16px',
            },
            flip: {
                enabled: false,
            },
        });
    });

    it('should render with only a custom distance', () => {
        const component = shallowWithTheme(
            <Popover isOpen={true} distance={6} renderContent={(): string => 'Mock content'} />,
        );

        const popper = component.find(Popper);

        expect(popper.prop('modifiers')).toEqual({
            offset: {
                offset: '0, 6px',
            },
            flip: {
                enabled: false,
            },
        });
    });

    it('should close when clicked outside the popover window', () => {
        const clickMock = jest.fn();
        const callbackMap: { [key: string]: Function } = {};

        document.addEventListener = jest.fn((event, callback) => (callbackMap[event] = callback));

        const component = mountWithTheme(
            <Popover
                isOpen={true}
                distance={6}
                onClickOutside={clickMock}
                renderContent={(): string => 'Mock content'}
            />,
        );

        callbackMap.mousedown({
            target: document.createElement('div'),
        });

        component.update();

        expect(clickMock).toHaveBeenCalled();
    });

    it('should not break when onClickOutside is undefined and there is clicked outside the popover window', () => {
        const callbackMap: { [key: string]: Function } = {};

        document.addEventListener = jest.fn((event, callback) => (callbackMap[event] = callback));

        const component = mountWithTheme(
            <Popover isOpen={true} distance={6} renderContent={(): string => 'Mock content'} />,
        );

        callbackMap.mousedown({
            target: document.createElement('div'),
        });

        component.update();
    });

    it('should not break when clicked outside the closed popover', () => {
        const fn = (): void => {
            const callbackMap: { [key: string]: Function } = {};

            document.addEventListener = jest.fn((event, callback) => (callbackMap[event] = callback));

            const component = mountWithTheme(
                <Popover isOpen={false} distance={6} renderContent={(): string => 'Mock content'} />,
            );

            callbackMap.mousedown({ target: document.createElement('div') });

            component.update();
        };

        expect(fn).not.toThrow();
    });

    it('adds and removes eventListeners', () => {
        const component = mountWithTheme(
            <Popover isOpen={true} distance={6} renderContent={(): string => 'Mock content'} />,
        );
        component.unmount();

        /* tslint:disable */
        expect((global as any).addEventListener).toBeCalled();
        expect((global as any).removeEventListener).toBeCalled();
        /* tslint:enable */
    });
});

describe('PopoverBackground', () => {
    it('should render with defaults', () => {
        const component = shallowWithTheme(<PopoverBackground />);

        expect(toJson(component)).toMatchSnapshot();
    });
});

describe('PopoverAnchor', () => {
    it('should render inline', () => {
        const component = shallowWithTheme(<PopoverAnchor />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render block', () => {
        const component = shallowWithTheme(<PopoverAnchor stretch={true} />);

        expect(toJson(component)).toMatchSnapshot();
    });
});

describe('PopoverArrow', () => {
    it('should render at the bottom with shadow', () => {
        const component = shallowWithTheme(<PopoverArrow placement="bottom" shadow />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render at the top', () => {
        const component = shallowWithTheme(<PopoverArrow placement="top" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render on the left', () => {
        const component = shallowWithTheme(<PopoverArrow placement="left" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render on the right', () => {
        const component = shallowWithTheme(<PopoverArrow placement="right" />);

        expect(toJson(component)).toMatchSnapshot();
    });
});
