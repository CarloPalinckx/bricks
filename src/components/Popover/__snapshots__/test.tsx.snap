// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover should render with defaults 1`] = `
<styled.div
  innerRef={null}
  style=""
>
  <styled.div>
    Mock content
  </styled.div>
  <styled.div />
  <styled.div
    innerRef={null}
    placement=""
    style=""
  />
  <styled.div
    placement=""
    shadow={true}
    style=""
  />
</styled.div>
`;

exports[`Popover should render with defaults 2`] = `
<styled.div
  innerRef={null}
/>
`;

exports[`PopoverAnchor should render inline 1`] = `
.c0 {
  display: inline-block;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="c0"
/>
`;

exports[`PopoverArrow should render at the bottom with shadow 1`] = `
.c0 {
  position: absolute;
  width: 16px;
  height: 16px;
  background: #fff;
  top: -8px;
  -webkit-transform: scaleY(.75) rotate(45deg);
  -ms-transform: scaleY(.75) rotate(45deg);
  transform: scaleY(.75) rotate(45deg);
  box-shadow: 0 1px 12px rgba(0 ,0,0,.2);
  z-index: 1;
}

<div
  className="c0"
/>
`;

exports[`PopoverArrow should render at the top 1`] = `
.c0 {
  position: absolute;
  width: 16px;
  height: 16px;
  background: #fff;
  bottom: -8px;
  -webkit-transform: scaleY(.75) rotate(45deg);
  -ms-transform: scaleY(.75) rotate(45deg);
  transform: scaleY(.75) rotate(45deg);
  background: #fff;
  z-index: 3;
}

<div
  className="c0"
/>
`;

exports[`PopoverArrow should render on the left 1`] = `
.c0 {
  position: absolute;
  width: 16px;
  height: 16px;
  background: #fff;
  right: -8px;
  -webkit-transform: scaleX(.75) rotate(45deg);
  -ms-transform: scaleX(.75) rotate(45deg);
  transform: scaleX(.75) rotate(45deg);
  background: #fff;
  z-index: 3;
}

<div
  className="c0"
/>
`;

exports[`PopoverArrow should render on the right 1`] = `
.c0 {
  position: absolute;
  width: 16px;
  height: 16px;
  background: #fff;
  left: -8px;
  -webkit-transform: scaleX(.75) rotate(45deg);
  -ms-transform: scaleX(.75) rotate(45deg);
  transform: scaleX(.75) rotate(45deg);
  background: #fff;
  z-index: 3;
}

<div
  className="c0"
/>
`;

exports[`PopoverBackground should render with defaults 1`] = `
.c0 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 2;
  box-shadow: 0 1px 12px rgba(0,0,0,0.2);
  background: #fff;
  border-radius: 5px;
}

<div
  className="c0"
/>
`;
